### PR TITLE
Feature/make dll

### PR DESCRIPTION
**Pull request by itaymmguardicore**

### CODE REVIEW

File: mimikatz.sln [hunk 1]
```diff
@@ -124,4 +87,8 @@ Global
 		{96078881-DEC1-4EF5-A24A-A81AC007809D} = {294B51F3-90EF-4F0A-BB04-20321A513A4F}
 		{E9D1619F-D4A1-4AFA-B261-B01091EB8D56} = {294B51F3-90EF-4F0A-BB04-20321A513A4F}
 	EndGlobalSection
+	GlobalSection(SubversionScc) = preSolution
```
Comment (danielguardicore) [inline mimikatz.sln]: Uh? Why?

File: mimikatz/logon_data.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <globals.h>
+#include <Windows.h>
+#define WINDOWS_MAX_USERNAME_PASS_LENGTH (257)
+
+typedef struct Node Node;
```
Comment (danielguardicore) [inline mimikatz/logon_data.h]: Generally, either don't call it something generic like List or decouple. I'm pro the former but the latter means
that I'd split this into two files, one for the linked list and the other actual logic using it.

File: mimikatz/logon_data.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <globals.h>
```
Comment (danielguardicore) [inline mimikatz/logon_data.h:2]: Once you include globals you don't need to explictly include windows.

File: mimikatz/logon_data.h [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+#include <Windows.h>
+#define WINDOWS_MAX_USERNAME_PASS_LENGTH (257)
```
Comment (danielguardicore) [inline mimikatz/logon_data.h]: Just add this to the relevant base file such as utils.h

File: mimikatz/logon_data.c [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+	memcpy_s(logonData->username, sizeof(WCHAR) * WINDOWS_MAX_USERNAME_PASS_LENGTH, username, sizeof(WCHAR) * username_size);
+	
+	if (password == NULL)
+	{
+		wmemset(logonData->password, L'\0', WINDOWS_MAX_USERNAME_PASS_LENGTH);
+	}
+	else
+	{
+		memcpy_s(logonData->password, sizeof(WCHAR) * WINDOWS_MAX_USERNAME_PASS_LENGTH, password, sizeof(WCHAR) * password_size);
```
Comment (danielguardicore) [inline mimikatz/logon_data.c]: Why not use strcpy_s? If you know this is a string.

File: mimikatz/mimikatz.c [hunk 6]
```diff
@@ -231,62 +197,27 @@ NTSTATUS mimikatz_doLocal(wchar_t * input)
 	return status;
 }
 
-#ifdef _POWERKATZ
-__declspec(dllexport) wchar_t * powershell_reflective_mimikatz(LPCWSTR input)
+size_t CALLBACK collectEntries(HWND hwnd, HINSTANCE hinst, LPWSTR lpszCmdLine, int nCmdShow)
```
Comment (danielguardicore) [inline mimikatz/mimikatz.c]: Remove the parameters if they're unused. Identical comment for getEntry.

File: mimikatz/modules/sekurlsa/kuhl_m_sekurlsa.c [hunk 7]
```diff
@@ -4,6 +4,7 @@
 	Licence : https://creativecommons.org/licenses/by/4.0/
 */
 #include "kuhl_m_sekurlsa.h"
+#include "../../logon_data.h"
```
Comment (danielguardicore) [inline mimikatz/modules/sekurlsa/kuhl_m_sekurlsa.c]: Why isn't VS automatically finding this? why do we need a relative path.

File: mimikatz/modules/sekurlsa/kuhl_m_sekurlsa.c [hunk 8]
```diff
@@ -1071,6 +1078,11 @@ VOID kuhl_m_sekurlsa_genericCredsOutput(PKIWI_GENERIC_PRIMARY_CREDENTIAL mesCred
 							}
 							if(sid && (*(PBOOLEAN) (msvCredentials + pMSVHelper->offsetToisNtOwfPassword) || *(PBOOLEAN) (msvCredentials + pMSVHelper->offsetToisShaOwPassword)))
 								kuhl_m_dpapi_oe_credential_add(sid, NULL, *(PBOOLEAN) (msvCredentials + pMSVHelper->offsetToisNtOwfPassword) ? msvCredentials + pMSVHelper->offsetToNtOwfPassword : NULL, *(PBOOLEAN) (msvCredentials + pMSVHelper->offsetToisShaOwPassword) ? msvCredentials + pMSVHelper->offsetToShaOwPassword : NULL, NULL, NULL);
+
+							if ((lmHash != NULL) || (ntlmHash != NULL))
+							{
+								List_addItem(s_list, LogonData_create(username->Buffer, username->Length, NULL, 0, lmHash, ntlmHash));
```
Comment (danielguardicore) [inline mimikatz/modules/sekurlsa/kuhl_m_sekurlsa.c]: Split this into assignment and passing on like later on or be consistent.

File: mimikatz.sln (hunk 1 above)
Comment (itaymmguardicore) [reply mimikatz.sln]: Removed

File: mimikatz/logon_data.h (hunk 3 above)
Comment (itaymmguardicore) [reply mimikatz/logon_data.h:2]: Fixed

File: mimikatz/logon_data.h (hunk 2 above)
Comment (itaymmguardicore) [reply mimikatz/logon_data.h]: Fixed

File: mimikatz/mimikatz.c (hunk 6 above)
Comment (itaymmguardicore) [reply mimikatz/mimikatz.c]: Fixed

File: mimikatz/modules/sekurlsa/kuhl_m_sekurlsa.c (hunk 7 above)
Comment (itaymmguardicore) [reply mimikatz/modules/sekurlsa/kuhl_m_sekurlsa.c]: Fixed

File: mimikatz/logon_data.h (hunk 4 above)
Comment (itaymmguardicore) [reply mimikatz/logon_data.h]: Fixed

File: mimikatz/modules/sekurlsa/kuhl_m_sekurlsa.c (hunk 8 above)
Comment (itaymmguardicore) [reply mimikatz/modules/sekurlsa/kuhl_m_sekurlsa.c]: Fixed

File: mimikatz/logon_data.c (hunk 5 above)
Comment (itaymmguardicore) [reply mimikatz/logon_data.c]: Fixed

**Review by danielguardicore — APPROVED** [review]

**Pull request merged**